tabbar.cpp

```diff
@@ -172,12 +237,31 @@ void TabBar::contextMenuEvent(QContextMenuEvent *event)
     menu->addSeparator();
     Core::EditorManager::addNativeDirAndOpenWithActions(menu.data(), entry);
 
+    QAction* mouse_wheel = new QAction(tr("Mouse Wheel?"), this);
+    mouse_wheel->setCheckable(true);
+    mouse_wheel->setChecked(m_wheel);
+    menu->addAction(mouse_wheel);
+
+    connect(mouse_wheel, SIGNAL(triggered(bool)),
+                   this, SLOT(mouseWheelChange(bool)));
+
     menu->exec(mapToGlobal(event->pos()));
 }
 
-void TabBar::mouseReleaseEvent(QMouseEvent *event)
+void TabBar::mouseWheelChange(bool checked)
+{
+    m_wheel = checked;
+}
+
+void TabBar::wheelEvent(QWheelEvent *event)
+{
+    if(m_wheel)
+        QTabBar::wheelEvent(event);
+}
+
+void TabBar::mousePressEvent(QMouseEvent *event)
 {
     if (event->button() == Qt::MiddleButton)

```

Comment:
This has to be in the release event handler, otherwise it'd be not possible to press and move mouse out to cancel.